### PR TITLE
[buildkite] Add Windows packaging and platform support tests to periodic pipeline

### PR DESCRIPTION
**Pull request by brianseeders**
This PR adds the platform/packaging tests that are run on Windows to the new Buildkite periodic pipeline.

Linux jobs will be added in a separate PR.

See here for test builds: https://buildkite.com/elastic/brianseeders-elasticsearch-sandbox/builds?branch=buildkite-migration-periodic-windows

See also: https://elasticco.atlassian.net/jira/software/c/projects/ES/boards/297?modal=detail&selectedIssue=ES-6358

- Add windows pre-command hook for setting up env vars / etc
- Add pipeline definitions
- Add gnarly workaround for buildkite-agent hanging job issue
- Make `run-gradle.sh` also work with Windows (previously unix only)
- Fix some environmental things that only worked in Jenkins
- Fix ReaperService to kill child processes correctly
- Fix buildkite annotations by using `ProcessBuilder.start()`

### CODE REVIEW

File: .buildkite/hooks/pre-command.bat [hunk 1]
```diff
@@ -0,0 +1,22 @@
+@ECHO OFF
+
+FOR /F "tokens=* eol=#" %%i in ('type .ci\java-versions.properties') do set %%i
+
+SET JAVA_HOME=%USERPROFILE%\.java\%ES_BUILD_JAVA%
+SET JAVA11_HOME=%USERPROFILE%\.java\java11
+SET JAVA16_HOME=%USERPROFILE%\.java\openjdk16
```
Comment (mark-vieira) [inline .buildkite/hooks/pre-command.bat:7]: Hmm, I'm wondering why we'd need Java 16. I think we can ditch this.

File: build-tools-internal/src/main/groovy/elasticsearch.build-scan.gradle [hunk 2]
```diff
@@ -121,12 +121,12 @@ buildScan {
       buildScanPublished { scan ->
         // Attach build scan link as build metadata
         // See: https://buildkite.com/docs/pipelines/build-meta-data
-        ['buildkite-agent', 'meta-data', 'set', "build-scan-${System.getenv('BUILDKITE_JOB_ID')}", "${scan.buildScanUri}"].execute()
+        new ProcessBuilder('buildkite-agent', 'meta-data', 'set', "build-scan-${System.getenv('BUILDKITE_JOB_ID')}", "${scan.buildScanUri}").start()
```
Comment (mark-vieira) [inline build-tools-internal/src/main/groovy/elasticsearch.build-scan.gradle:124]: Should we also call `waitFor()` here?

**Review by mark-vieira — APPROVED** [review]
Couple comments but otherwise LGTM.

File: build-tools-internal/src/main/groovy/elasticsearch.build-scan.gradle (hunk 2 above)
Comment (brianseeders) [reply build-tools-internal/src/main/groovy/elasticsearch.build-scan.gradle]: Oops, yep, thanks.

**Pull request merged**